### PR TITLE
[Gluten-137]Fix arrow installation order and Velox installation directory

### PR DESCRIPTION
**Pull request by weiting-chen**
## What changes were proposed in this pull request?

1. Fix arrow installation order
2. Change Velox installation directory

## How was this patch tested?
Pass compile testing in my local server

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,14 +48,15 @@
     <build_cpp>OFF</build_cpp>
     <cpp_tests>OFF</cpp_tests>
     <build_arrow>OFF</build_arrow>
-    <static_arrow>OFF</static_arrow>
     <arrow_root>/usr/local</arrow_root>
+    <arrow.bfs.dir>${project.basedir}/../tools/build/arrow_install</arrow.bfs.dir>
+    <static_arrow>OFF</static_arrow>
     <build_protobuf>ON</build_protobuf>
     <build_jemalloc>ON</build_jemalloc>
```
Comment (zhouyuan) [inline pom.xml:55]: can you also disable jemalloc? it's not used actually

Comment (weiting-chen) [reply pom.xml]: Done

File: tools/build_arrow.sh [hunk 2]
```diff
@@ -99,7 +102,7 @@ cmake -DARROW_BUILD_STATIC=OFF \
         -DARROW_SIMD_LEVEL=AVX2 \
         -DARROW_RUNTIME_SIMD_LEVEL=MAX \
         -DARROW_DEPENDENCY_SOURCE=BUNDLED \
-        -DCMAKE_INSTALL_PREFIX=/tmp/arrow_install.8 \
+        -DCMAKE_INSTALL_PREFIX=$ARROW_INSTALL_DIR \
```
Comment (zhztheplayer) [inline tools/build_arrow.sh:105]: 👍

File: tools/build_velox.sh [hunk 3]
```diff
@@ -59,7 +59,7 @@ if [ $BUILD_VELOX_FROM_SOURCE == "ON" ]; then
     sed -i '/protobuf-compiler/d' scripts/setup-ubuntu.sh
     sed -i 's/^  liblzo2-dev.*/  liblzo2-dev/g' scripts/setup-ubuntu.sh
     sed -i 's/^  ninja -C "${BINARY_DIR}" install/  sudo ninja -C "${BINARY_DIR}" install/g' scripts/setup-ubuntu.sh
-    sed -i '/^function install_folly.*/i function install_pb {\n  github_checkout protocolbuffers/protobuf v3.13.0\n  git submodule update --init --recursive\n  ./autogen.sh\n  ./configure\n  make\n  make check\n  sudo make install\n sudo ldconfig\n}\n' scripts/setup-ubuntu.sh
+    sed -i '/^function install_folly.*/i function install_pb {\n  github_checkout protocolbuffers/protobuf v3.13.0\n  git submodule update --init --recursive\n  ./autogen.sh\n  ./configure CFLAGS=-fPIC CXXFLAGS=-fPIC\n  make\n  make check\n  sudo make install\n sudo ldconfig\n}\n' scripts/setup-ubuntu.sh
     sed -i '/^  run_and_time install_folly/i \ \ run_and_time install_pb' scripts/setup-ubuntu.sh
```
Comment (zhouyuan) [inline tools/build_velox.sh:63]: make -j${NPROC} ?

**Pull request merged**